components/gitpod-cli/cmd/rebuild.go

```diff
@@ -0,0 +1,241 @@
+	}
+
+	tmpDir, err := os.MkdirTemp("", "gp-rebuild-*")
+	if err != nil {
+		event.Set("ErrorCode", utils.SystemErrorCode)
+		return err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	gitpodConfig, err := utils.ParseGitpodConfig(wsInfo.CheckoutLocation)
+	if err != nil {
+		fmt.Println("The .gitpod.yml file cannot be parsed: please check the file and try again")
+		fmt.Println("")
+		fmt.Println("For help check out the reference page:")
+		fmt.Println("https://www.gitpod.io/docs/references/gitpod-yml#gitpodyml")
+		event.Set("ErrorCode", utils.RebuildErrorCode_MalformedGitpodYaml)
+		return err
+	}
+
+	if gitpodConfig == nil {
+		fmt.Println("To test the image build, you need to configure your project with a .gitpod.yml file")
+		fmt.Println("")
+		fmt.Println("For a quick start, try running:\n$ gp init -i")
+		fmt.Println("")
+		fmt.Println("Alternatively, check out the following docs for getting started configuring your project")
+		fmt.Println("https://www.gitpod.io/docs/configure#configure-gitpod")
+		event.Set("ErrorCode", utils.RebuildErrorCode_MissingGitpodYaml)
+		return err
+	}
+
+	var baseimage string
+	switch img := gitpodConfig.Image.(type) {
+	case nil:
+		baseimage = ""
+	case string:
+		baseimage = "FROM " + img
+	case map[interface{}]interface{}:
+		dockerfilePath := filepath.Join(wsInfo.CheckoutLocation, img["file"].(string))
+
+		if _, err := os.Stat(dockerfilePath); os.IsNotExist(err) {
+			fmt.Println("Your .gitpod.yml points to a Dockerfile that doesn't exist: " + dockerfilePath)
+			event.Set("ErrorCode", utils.RebuildErrorCode_DockerfileNotFound).Send(ctx)
+			return err
+		}
+		dockerfile, err := os.ReadFile(dockerfilePath)
+		if err != nil {
+			event.Set("ErrorCode", utils.RebuildErrorCode_DockerfileCannotRead)
+			return err
+		}
+		if string(dockerfile) == "" {
+			fmt.Println("Your Gitpod's Dockerfile is empty")
+			fmt.Println("")
+			fmt.Println("To learn how to customize your workspace, check out the following docs:")
+			fmt.Println("https://www.gitpod.io/docs/configure/workspaces/workspace-image#use-a-custom-dockerfile")
+			fmt.Println("")
+			fmt.Println("Once you configure your Dockerfile, re-run this command to validate your changes")
+			event.Set("ErrorCode", utils.RebuildErrorCode_DockerfileEmpty)
+			return err
+		}
+		baseimage = "\n" + string(dockerfile) + "\n"
+	default:
+		fmt.Println("Check your .gitpod.yml and make sure the image property is configured correctly")
+		event.Set("ErrorCode", utils.RebuildErrorCode_MalformedGitpodYaml)
+		return err
+	}
+
+	if baseimage == "" {
+		fmt.Println("Your project is not using any custom Docker image.")
+		fmt.Println("Check out the following docs, to know how to get started")
+		fmt.Println("")
+		fmt.Println("https://www.gitpod.io/docs/configure/workspaces/workspace-image#use-a-public-docker-image")
+		event.Set("ErrorCode", utils.RebuildErrorCode_NoCustomImage)
+		return err
+	}
+
+	err = os.WriteFile(filepath.Join(tmpDir, "Dockerfile"), []byte(baseimage), 0644)
```

Comment:
@andreafalzetti @felladrin is not it bogus? 🤔 

We should build in the context of /workspace because Dockerfile can make usage of COPY statements.

Reply:
I think we can use the current working direction to run docker command but then pass `-f` which pints to the temp dockerfile?